lib/vanagon/project.rb

```diff
@@ -494,6 +494,21 @@ def get_postinstall_actions(pkg_state)
       end
     end
 
+    # Collects the postinstall packaging actions that must exit successfully for the project and it's components
+    #  for the specified packaging state
+    #
+    # @param pkg_state [String] the package state we want to run the given scripts for.
+    #   Can be one of 'install' or 'upgrade'
+    # @return [String] string of Bourne shell compatible scriptlets to execute during the postinstall
+    #   phase of packaging during the state of the system defined by pkg_state (either install or upgrade)
+    def get_postinstall_actions(pkg_state)
+      scripts = components.flat_map(&:postinstall_required_actions).compact.select { |s| s.pkg_state.include? pkg_state }.map(&:scripts)
+      if scripts.empty?
+        return ': no postinstall required scripts provided'
+      else
```

Comment:
Style only: `else` is redundant. Favor a guardian clause:

```ruby
return ': no postinstall required scripts provided' if scripts.empty?

[return] scripts.join("\n")
```

The 2nd return is optional depending on your aesthetic preferences.